### ISSUE TITLE
Infinite scroll issue

### ISSUE THREAD

**Issue by binoclard** [imacrayon/alpine-ajax#28]
Hi,

First, thank you for your work.

I am having trouble with infinite scroll implementation. 

As it is, I get a `Target #pagination not found in AJAX response.`error logged, and the same page tow loaded again and again as I scroll…

```antlers
<section id="articles"
    class="grid items-start sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 gap-6"
    x-data x-masonry.wait.100 x-arrange="append">

        {{ collection from="journal" sort="date:desc" paginate="25" as="billet" }}

            {{ billet }}
                {{ partial src="brick" preset="{view:glide_presets | shuffle | first}" }}
            {{ /billet }}

            {{ paginate }}
                <div class="" id="pagination" x-init x-intersect="$ajax('{{ next_page }}', {events: 'true'})"
                    x-target="articles pagination" @ajax:success="$dispatch('reload:masonry')">
                </div>
            {{ /paginate }}

        {{ /collection }}
</section>
```

I am using Statamic to display entries, and the way the pagination works, the pagination loop has to be inside the collection entries loop. Or, I have noticed that in your example https://alpine-ajax.js.org/examples/infinite-scroll/, the pagination is outside the target table body where the new elements are loaded and added.
I suppose this explains that, perhaps? But I do not understand why the plugin could not find the #pagination, wherever it is in the markup?

If I duplicate the `{{ collection }}` loop, once with the entries, and once with the pagination, outside of the target, it kinda works, but feels "messy".

Any inputs/ideas?

Thanks in advance for your thoughts.

**Comment by imacrayon** [comment]
Thanks for reaching out Olivier, 

The "#pagination not found" error does seem like a bug, so I'll look into that.

However, there is an issue with the way your markup is structured: Your pagination `div` is nested inside the `section` where content is going to be appended, so when a new page is requested, the `#pagination` on the new page will also get appended to your list of articles, so you'll end up with two `div`s having `id="pagination"` on the same page. Two elements with the same `id` is going to start causing problems.

If you can keep the `#articles` and `#pagination` containers separated, you can avoid duplicating the `#pagination` container on every requests. I think Statamic will allow you to build your template like this:

```html
{{ collection from="journal" sort="date:desc" paginate="25" as="billet" }}
  <section id="articles"
      class="grid items-start sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 gap-6"
      x-data x-masonry.wait.100 x-arrange="append">
      {{ billet }}
          {{ partial src="brick" preset="{view:glide_presets | shuffle | first}" }}
      {{ /billet }}
  </section>

  {{ paginate }}
      <div class="" id="pagination" x-init x-intersect="$ajax('{{ next_page }}', {events: 'true'})"
          x-target="articles pagination" @ajax:success="$dispatch('reload:masonry')">
      </div>
  {{ /paginate }}
{{ /collection }}

```

Using `x-arrange="append"` gets a little wild, and I think Alpine AJAX could have some better error handling around these situations, so thanks for

**Comment by binoclard** [comment]
Thank you! Of course! I can't believe I searched for convoluted solutions when it was plain and obvious.

FYI, the error is not there anymore, so it is maybe not a bug, just a consequence of the nested markup.

I had to tinker a bit to end with `x-intersect:leave.once="$ajax('{{ next_page }}'` but it seems to work fine, even if my browser and computer do not seem to like it much, and hyperventilate 😅

Thanks again.
